### PR TITLE
Intrinsic and Language docs

### PR DESCRIPTION
**Pull request by Vandesm14**
- Updated all docs to the new language spec, including examples.
- Added docs for intrinsics
- Added docs on loops
- Added docs for the debugger (installation & running only)
- Updated the README a bit

### CODE REVIEW

**Comment by Vandesm14** [comment]
@leonskidev How is this so far? Do you have any other inclusions besides descriptions and examples for each intrinsic?

File: docs/src/reference/builtins.md [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+
+**Signature:** `([a: int] [b: int] -- int)`
+
+**Equivalent Rust:** `a / b`
+
+### Remainder (`%`)
+
+**Signature:** `([a: int] [b: int] -- int)`
+
+**Equivalent Rust:** `a & b`
```
Comment (leonskij) [inline docs/src/reference/builtins.md]: Stack uses wrapping arithmetic whereas Rust has bounds checking on arithmetic in safe modes. I'm not sure if it necessarily matters for the purposes of these examples though?

Comment (leonskij) [inline docs/src/reference/builtins.md]: The *'Equivalent Rust'* should be *'a % b'* here; also see the previous comment.

File: docs/src/reference/builtins.md [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+
+**Signature:** `([a] --)`
+
+**Equivalent Rust:** `println!("{}", a)`
+
+### Pretty (`pretty`)
+
+**Signature:** `([a] --)`
+
+**Equivalent Rust:** `println!("{:#}", val)`
```
Comment (leonskij) [inline docs/src/reference/builtins.md]: Is it worth mentioning these? Given that they will be part of the standard-library soon-ish.

File: docs/src/introduction/scopes.md [hunk 3]
```diff
@@ -2,27 +2,29 @@
...
 
 ## What is a Scope?
 
-Similar to other languages, a scope is a collection of symbol-expression pairs. When a variable is defined, it is added to the current scope. When a variable is called, it is looked up in the current scope (by the [purification](../glossary#purification) step).
+Similar to other languages, a scope is a collection of symbol-expression pairs. When a variable is defined, it is added to the current scope. When a variable is called, it is looked up in the current scope (by the [purification](stack.md#purification) step).
 
 ## The Main Scope
 
-The main scope is the top-level of your program. It is the first scope that is created when you start Stack. It is the parent of all subsequent scopes. It acts as a global scope, though some behavior is isolated when inside of a function (more on that later).
+The main scope is the top-level of your program. It is the first scope that is created when you start Stack. It is the outermost scope and acts as a global scope.
```
Comment (leonskij) [inline docs/src/introduction/scopes.md]: The last two sentences can be connected with *'and'*.

File: docs/theme/LICENSE
Comment (leonskij) [inline docs/theme/LICENSE:1]: This should probably be renamed to *`HIGHLIGHT-LICENCE`*, just for distinction.

**Pull request merged**